### ISSUE TITLE
com.hazelcast.map.LocalMapStatsTest_WhenMigration.testHitsGenerated_newNode

### ISSUE THREAD

**Issue by bpoplawski** [hazelcast/hazelcast#21322]
_5.1.z_ (commit ce56585fff5f4ae636ea91d2c9357836a3c0c39e )

Failed on Zing JDK8: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-ZingJDK8/64/consoleFull

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<100> but was:<63>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.map.LocalMapStatsTest_WhenMigration.lambda$testHitsGenerated_newNode$0(LocalMapStatsTest_WhenMigration.java:69)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1225)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1327)
	at com.hazelcast.map.LocalMapStatsTest_WhenMigration.testHitsGenerated_newNode(LocalMapStatsTest_WhenMigration.java:65)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: testHitsGenerated_nodeCrash in 0.590 seconds.
20:29:48,263 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:29:53,266 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:29:58,265 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:03,265 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:08,265 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:13,265 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:18,265 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:23,266 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:28,267 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:33,269 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:38,268 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:43,269 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:48,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:53,270 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:30:58,270 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:03,271 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:08,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:13,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:18,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:23,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:28,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:33,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:38,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:43,272 DEBUG |testHitsGenerated_newNode| - [JobRepository] hz.elated_franklin.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Job cleanup took 0ms
20:31:43,479  INFO |testHitsGenerated_newNode| - [LifecycleService] Thread-56577 - [127.0.0.1]:5702 [dev] [5.1.2-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
20:31:43,479  WARN |testHitsGenerated_newNode| - [Node] Thread-56577 - [127.0.0.1]:5702 [dev] [5.1.2-SNAPSHOT] Terminating forcefully...
20:31:43,479  INFO |testHitsGenerated_newNode| - [Node] Thread-56577 - [127.0.0.1]:5702 [dev] [5.1.2-SNAPSHOT] Shutting down connection manager...
20:31:43,479  INFO |testHitsGenerated_newNode| - [MockServer] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Removed connection to endpoint: [address=[127.0.0.1]:5702, uuid=20f0ac2b-ef42-4057-8650-4f2927d87fa5], connection: MockConnection{localEndpoint=[address=[127.0.0.1]:5701, uuid=5e32acc8-4e3f-43c0-b52d-7fbb1069d72e], remoteEndpoint=[address=[127.0.0.1]:5702, uuid=20f0ac2b-ef42-4057-8650-4f2927d87fa5], alive=false}
20:31:43,479  INFO |testHitsGenerated_newNode| - [MockServer] Thread-56577 - [127.0.0.1]:5702 [dev] [5.1.2-SNAPSHOT] Removed connection to endpoint: [address=[127.0.0.1]:5701, uuid=5e32acc8-4e3f-43c0-b52d-7fbb1069d72e], connection: MockConnection{localEndpoint=[address=[127.0.0.1]:5702, uuid=20f0ac2b-ef42-4057-8650-4f2927d87fa5], remoteEndpoint=[address=[127.0.0.1]:5701, uuid=5e32acc8-4e3f-43c0-b52d-7fbb1069d72e], alive=false}
20:31:43,479  INFO |testHitsGenerated_newNode| - [MembershipManager] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Removing Member [127.0.0.1]:5702 - 20f0ac2b-ef42-4057-8650-4f2927d87fa5
20:31:43,480  INFO |testHitsGenerated_newNode| - [PartitionStateManager] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Storing snapshot of partition assignments while removing UUID 20f0ac2b-ef42-4057-8650-4f2927d87fa5
20:31:43,480  INFO |testHitsGenerated_newNode| - [ClusterService] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5701 - 5e32acc8-4e3f-43c0-b52d-7fbb1069d72e this
]

20:31:43,480  INFO |testHitsGenerated_newNode| - [Node] Thread-56577 - [127.0.0.1]:5702 [dev] [5.1.2-SNAPSHOT] Shutting down node engine...
20:31:43,480  INFO |testHitsGenerated_newNode| - [TransactionManagerService] hz.elated_franklin.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 20f0ac2b-ef42-4057-8650-4f2927d87fa5
20:31:43,496  INFO |testHitsGenerated_newNode| - [MigrationManager] hz.elated_franklin.migration - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Partition balance is ok, no need to repartition.
20:31:43,509  INFO |testHitsGenerated_newNode| - [NodeExtension] Thread-56577 - [127.0.0.1]:5702 [dev] [5.1.2-SNAPSHOT] Destroying node NodeExtension.
20:31:43,509  INFO |testHitsGenerated_newNode| - [Node] Thread-56577 - [127.0.0.1]:5702 [dev] [5.1.2-SNAPSHOT] Hazelcast Shutdown is completed in 30 ms.
20:31:43,509  INFO |testHitsGenerated_newNode| - [LifecycleService] Thread-56577 - [127.0.0.1]:5702 [dev] [5.1.2-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
20:31:43,509  INFO |testHitsGenerated_newNode| - [LifecycleService] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
20:31:43,509  WARN |testHitsGenerated_newNode| - [Node] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Terminating forcefully...
20:31:43,510  INFO |testHitsGenerated_newNode| - [Node] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Shutting down connection manager...
20:31:43,510  INFO |testHitsGenerated_newNode| - [Node] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Shutting down node engine...
20:31:43,515  INFO |testHitsGenerated_newNode| - [NodeExtension] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Destroying node NodeExtension.
20:31:43,516  INFO |testHitsGenerated_newNode| - [Node] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] Hazelcast Shutdown is completed in 7 ms.
20:31:43,516  INFO |testHitsGenerated_newNode| - [LifecycleService] Thread-56577 - [127.0.0.1]:5701 [dev] [5.1.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after testHitsGenerated_newNode(com.hazelcast.map.LocalMapStatsTest_WhenMigration): BuildInfo{version='5.1.2-SNAPSHOT', build='20220426', buildNumber=20220426, revision=ce56585, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testHitsGenerated_newNode(com.hazelcast.map.LocalMapStatsTest_WhenMigration):'
20:29:40, accumulated pauses: 4907 ms, max pause: 25 ms, pauses over 1000 ms: 0
20:29:45, accumulated pauses: 4901 ms, max pause: 18 ms, pauses over 1000 ms: 0
20:29:50, accumulated pauses: 4894 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:29:55, accumulated pauses: 4894 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:30:00, accumulated pauses: 4904 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:30:05, accumulated pauses: 4890 ms, max pause: 12 ms, pauses over 1000 ms: 0
20:30:10, accumulated pauses: 4902 ms, max pause: 14 ms, pauses over 1000 ms: 0
20:30:15, accumulated pauses: 4890 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:30:20, accumulated pauses: 4899 ms, max pause: 12 ms, pauses over 1000 ms: 0
20:30:25, accumulated pauses: 4897 ms, max pause: 15 ms, pauses over 1000 ms: 0
20:30:30, accumulated pauses: 4899 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:30:35, accumulated pauses: 4897 ms, max pause: 14 ms, pauses over 1000 ms: 0
20:30:40, accumulated pauses: 4889 ms, max pause: 11 ms, pauses over 1000 ms: 0
20:30:45, accumulated pauses: 4903 ms, max pause: 14 ms, pauses over 1000 ms: 0
20:30:50, accumulated pauses: 4897 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:30:55, accumulated pauses: 4890 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:31:00, accumulated pauses: 4898 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:31:05, accumulated pauses: 4903 ms, max pause: 12 ms, pauses over 1000 ms: 0
20:31:10, accumulated pauses: 4893 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:31:15, accumulated pauses: 4899 ms, max pause: 13 ms, pauses over 1000 ms: 0
20:31:20, accumulated pauses: 4890 ms, max pause: 14 ms, pauses over 1000 ms: 0
20:31:25, accumulated pauses: 4899 ms, max pause: 17 ms, pauses over 1000 ms: 0
20:31:30, accumulated pauses: 4896 ms, max pause: 14 ms, pauses over 1000 ms: 0
20:31:35, accumulated pauses: 4897 ms, max pause: 12 ms, pauses over 1000 ms: 0
20:31:40, accumulated pauses: 3434 ms, max pause: 13 ms, pauses over 1000 ms: 0

```

</details>

**Comment by olukas** [comment]
Failed on `5.1.z` on Openj9-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-Openj9-17/27/testReport/com.hazelcast.map/LocalMapStatsTest_WhenMigration/testHitsGenerated_newNode/

**Comment by olukas** [comment]
Failed on `5.1.z` on ZuluJDK11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-ZuluJDK11/54/testReport/com.hazelcast.map/LocalMapStatsTest_WhenMigration/testHitsGenerated_newNode/

**Comment by ahmetmircik** [comment]
@arodionov could you also please backport your fix, both OS and EE ones, to 5.1 and 5.0: https://github.com/hazelcast/hazelcast/pull/22138?
